### ISSUE TITLE
datasets.MNIST hans```

### ISSUE THREAD

**Issue by demongolem-biz** [pytorch/pytorch#90191]
### 🐛 Describe the bug

Pretty simple really.  The code below tries to get code from yann.lecun.com.  From going there myself, 3 of the 4 gzip files are downloadable but one doesn't download correctly or might be corrupt.  This file is train-labels-idx1-ubyte.gz.

I have identified an alternative site to get a good version of this file.  However if I run the below code, the code hangs when trying to fetch the above code.  Hanging should not be the correct operation.  Preferred to visit another location automatically through code or to give up on downloading the file altogether and continue.

I suppose download=False might get me past this. but it does not seem to be too satisfactory of a solution.

```
train_loader = torch.utils.data.DataLoader(
    datasets.MNIST(
        "../data", train=True, download=True, transform=transforms.Compose(
            [transforms.ToTensor(), transforms.Normalize((0.1307,), (0.3081,))]),), 
    batch_size=args.batch_size,
    shuffle=True,
    **kwargs)
```

### Versions

torch is 1.13.0, torchvision is 0.14.0
